tools/project.py

```diff
@@ -155,6 +155,12 @@ def __init__(self) -> None:
         self.custom_build_steps: Optional[Dict[str, List[Dict[str, Any]]]] = (
             None  # Custom build steps, types are ["pre-compile", "post-compile", "post-link", "post-build"]
         )
+        self.make_clangd_config: bool = (
+            False  # Generate compile_commands.json for clangd
+        )
```

Comment:
I was thinking we could turn it on by default

Reply:
I'm a little unsure about enabling it by default, partially because it's an extraneous file for projects that don't end up using it, but moreso because currently, when enabled it accounts for more than half of the time it takes to configure. Certainly some performance improvements to be had lol

Reply:
Really? I’m surprised, it seems like it’s doing about the same amount of work as the other two steps.

Reply:
I would be interested to see what it's doing in a [profiler](https://docs.python.org/3/library/profile.html). I can't guess what's slowing it down either.